### ISSUE TITLE
macOS: Building stabilization/2111RTE branch build fails

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#6407]
**Describe the bug**
CMake fails to build stabilization/2111RTE branch on macOS due to the following error:

```
CompileC /Users/lumberpro6/o3de/osx_build/External/ScriptCanvas-de3db492/Code/O3DE.build/profile/ScriptCanvas.Editor.Static.build/Objects-normal/x86_64/unity_0_cxx.o 
...

** BUILD FAILED **

The following build commands failed:
	CompileC /Users/lumberpro6/o3de/osx_build/External/ScriptCanvas-de3db492/Code/O3DE.build/profile/ScriptCanvas.Editor.Static.build/Objects-normal/x86_64/unity_0_cxx.o /Users/lumberpro6/o3de/osx_build/External/ScriptCanvas-de3db492/Code/CMakeFiles/ScriptCanvas.Editor.Static.dir/Unity/unity_0_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
Elapsed time: 4215 s. (time), 0.168346 s. (clock)
```

The issue occured on every tested macOS machine which were:

* Macbook Pro 2017 (Intel i7)
* Macbook Pro 2019 (Intel i9)
* Macbook Air 2020 (M1)

For further information please refer to the attached building logs.

**Steps to reproduce**

1. Obtain the stabilization/2111RTE branch build via Github.
2. Create a new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
3. Obtain python packages with the _get_python.sh_ script located in the _o3de/python_ directory.
4. Create the osx_build folder under the o3de location.
5. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
```
6. Configure and generate the Xcode solution with the following command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`

7. Build Mac binaries with the following command:
`cmake -E time cmake --build . --target ALL_BUILD --config profile`

**Expected behavior**
Building process finishes successfully without errors.

**Actual behavior**
Building fails.

**Found in Branch**
Stabilization/2111RTE (a2bb968)

**Desktop/Device (please complete the following information):**

- Device: Mac
- OS: macOS
- Version: Big Sur 11.3.1
- CPU: Intel Core i7 2,8 GHz
- GPU: Radeon Pro 555
- Memory: 16GB

**Additional context**
[Stabilization_2111RTE build fail log Macbook Pro 2017.txt](https://github.com/o3de/o3de/files/7712035/Stabilization_2111RTE.build.fail.log.Macbook.Pro.2017.txt)

[Stabilization_2111RTE build fail log macpro 2019.txt](https://github.com/o3de/o3de/files/7712038/Stabilization_2111RTE.build.fail.log.macpro.2019.txt)

**Comment by moraaar** [comment]
This bug is fixed in development branch (https://github.com/o3de/o3de/pull/6237) it has to be cherry picked into RTE branch.

**Comment by LB-MateuszSzweda** [comment]
Verified as merged to and fixed on Stabilization/2111RTE (d7dc257).
